addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1053,16 +1072,30 @@ options.registry.OptionsTab = options.Class.extend({
             // comparaison with BS grays. However the system supports any
             // gray palette so we base our hue and extra saturation displayed
             // values on one gray, let's say the middle one: gray-500.
-            const gray = weUtils.getCSSVariableValue('500');
-            const grayRGB = ColorpickerWidget.convertCSSColorToRgba(gray);
-            const hsl = ColorpickerWidget.convertRgbToHsl(grayRGB.red, grayRGB.green, grayRGB.blue);
-
-            const baseGray = weUtils.getCSSVariableValue('base-500');
-            const baseGrayRGB = ColorpickerWidget.convertCSSColorToRgba(baseGray);
-            const baseHSL = ColorpickerWidget.convertRgbToHsl(baseGrayRGB.red, baseGrayRGB.green, baseGrayRGB.blue);
-
-            this.grayParams['gray-hue'] = hsl.hue;
-            this.grayParams['gray-extra-saturation'] = hsl.saturation - baseHSL.saturation;
+            const grey = _.range(100, 1000, 100).map(id => {
+                const gray = weUtils.getCSSVariableValue(`${id}`);
+                const grayRGB = ColorpickerWidget.convertCSSColorToRgba(gray);
+                const hsl = ColorpickerWidget.convertRgbToHsl(grayRGB.red, grayRGB.green, grayRGB.blue);
+
+                const baseGray = weUtils.getCSSVariableValue(`base-${id}`);
+                const baseGrayRGB = ColorpickerWidget.convertCSSColorToRgba(baseGray);
+                const baseHSL = ColorpickerWidget.convertRgbToHsl(baseGrayRGB.red, baseGrayRGB.green, baseGrayRGB.blue);
+
+                return ({
+                    id,
+                    baseHSL,
+                    hsl
+                })
+            })
+            const [first] = grey;
+            const maxValue = grey.reduce((gray, value) => {
+                return gray.hsl.saturation > value.hsl.saturation ? gray : value
+            }, first);
+            const minValue = grey.reduce((gray, value) => {
+                return gray.hsl.saturation < value.hsl.saturation ? gray : value
+            }, first);
+            this.grayParams['gray-hue'] = first.hsl.hue;
+            this.grayParams['gray-extra-saturation'] = Math.min((maxValue.hsl.saturation - maxValue.baseHSL.saturation), (minValue.hsl.saturation - minValue.baseHSL.saturation));
```

Comment:
Is using Math.min the right way? :thinking: Let's say the min and max saturation of base gray is 5 and 20. If I use 90 extra saturation to reach 95 and 100, I have here `Math.min(100 - 20, 95 - 5)` = min(80, 90) = 80 and so I don't find my 90 back.

Also, I don't think you need the recompute the saturation of base grays since you did it to know about the min and max of the range element. It should be accessible via the `params` argument you receive in that function.